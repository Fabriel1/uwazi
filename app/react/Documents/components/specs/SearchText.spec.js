import React from 'react';
import {shallow} from 'enzyme';

import {SearchText} from '../SearchText.js';

describe('SearchText', () => {
  let component;
  //let instance;
  let props;

  let render = () => {
    component = shallow(<SearchText {...props}/>);
    //instance = component.instance();
  };

  beforeEach(() => {
    props = {
      snippets: [
        'first <b>snippet 1</b> found',
        'second <b>snippet 2</b> found',
        'third <b>snippet 3</b> found'
      ]
    };
    render();
  });

  it('should render all snippets with dangerouslySetInnerHTML', () => {
    let snippets = component.find('li');
    expect(snippets.length).toBe(3);
    expect(snippets.at(0).props().dangerouslySetInnerHTML).toEqual({__html: props.snippets[0]});
    expect(snippets.at(1).props().dangerouslySetInnerHTML).toEqual({__html: props.snippets[1]});
    expect(snippets.at(2).props().dangerouslySetInnerHTML).toEqual({__html: props.snippets[2]});
  });
});
