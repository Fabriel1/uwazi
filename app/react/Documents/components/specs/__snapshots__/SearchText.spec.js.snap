// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchText blankState when there is no search term should render a blank state message 1`] = `
<div
  className="blank-state"
>
  <Component
    icon="search"
  />
  <h4>
    Search text
  </h4>
  <p>
    Search text description
  </p>
</div>
`;

exports[`SearchText blankState when there is search term should render a no matching text message 1`] = `
<div
  className="blank-state"
>
  <Component
    icon="search"
  />
  <h4>
    No text match
  </h4>
  <p>
    No text match description
  </p>
</div>
`;
